### PR TITLE
Enable Conversion Warnings

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Since @ZehMatt keeps requesting them.

### CODE REVIEW

File: cmake/OpenLocoUtility.cmake [hunk 1]
```diff
@@ -63,7 +63,6 @@ function(loco_target_compile_link_flags TARGET)
         /wd4200                  #   4200: nonstandard extension used : zero-sized array in struct/union
         /wd4201                  #   4201: nonstandard extension used : nameless struct/union
         /wd4244                  #   4244: 'argument' : conversion from 'type1' to 'type2', possible loss of data
```
Comment (duncanspumpkin) [inline cmake/OpenLocoUtility.cmake:65]: This one requires tons of work to enable unfortunately.

**Review by LeftofZen — COMMENTED** [review]
+1 for this PR, more warnings is good

**Review by LeftofZen — APPROVED** [review]
+1 for this PR, more warnings is good

**Review by ZehMatt — APPROVED** [review]
Good start

**Pull request merged**